### PR TITLE
Make checkpoint signatures available from checkpoint tracker.

### PR DESCRIPTION
**Pull request by arai-fortanix**
This updates the transparency log checkpoint tracker to make the checkpoint's signature(s) available for client code that wants to verify the signature itself.

### CODE REVIEW

**Comment by roger2hk** [comment]
```
Already have image: gcr.io/trillian-opensource-ci/testbase
+ ./scripts/presubmit.sh --no-build --no-linters --no-generate --no-docker --no-actions
Building serverless wasm ====================
# github.com/google/trillian-examples/serverless/experimental/wasm
serverless/experimental/wasm/main.go:92:17: assignment mismatch: 3 variables but client.FetchCheckpoint returns 4 values
serverless/experimental/wasm/main.go:248:17: assignment mismatch: 3 variables but client.FetchCheckpoint returns 4 values
```

File: serverless/integration/integration_test.go [hunk 1]
```diff
@@ -93,6 +96,32 @@ func RunIntegration(t *testing.T, s log.Storage, f client.Fetcher, lh *rfc6962.H
 			t.Fatalf("Failed to update tracked log state: %q", err)
 		}
 		newCheckpoint := lst.LatestConsistent
+		// Verify that the returned checkpoint note is as expected.
```
Comment (AlCutter) [inline serverless/integration/integration_test.go:99]: Can you just save the `cpNote` (and perhaps `cpNoteSigned`) from L84/L85 and compare them with what you get back from `Update`?

It seems like a lot of the testing below is recovering things which should already be covered by tests in the `log` package (around the correct form of a `log.Checkpoint`, etc.)

**Review by AlCutter — COMMENTED** [review]
Also, it looks like there might be more references to the updated function to fix - see Roger's comment above.

**Review by AlCutter — COMMENTED** [review]
Sorry for the slow response, just one little nit inline.

**Review by AlCutter — APPROVED** [review]

**Pull request merged**